### ISSUE TITLE
torch.split_with_sizes is not documented

### ISSUE THREAD

**Issue by kshitij12345** [pytorch/pytorch#58181]
![Screenshot from 2021-05-13 01-35-59](https://user-images.githubusercontent.com/19503980/118037551-a1b8c900-b38b-11eb-8515-83f74946a3f2.png)

The function is available and tested. Also the name does not start with a leading underscore so I assume it is meant to be public. Looks like this was just a miss.
https://github.com/pytorch/pytorch/blob/c9eb381aac886ea39efa4ad042fa126102974905/torch/testing/_internal/common_methods_invocations.py#L6382-L6384

**Comment by mruberry** [comment]
cc @albanD 

Before documenting split_with_sizes we should check its Python Array API or NumPy consistency

**Comment by kshitij12345** [comment]
AFAIK, it does not have a direct name equivalent in NumPy (semantically I think it is similar to np.split with list or tuple for `indices_or_sections`)

https://numpy.org/doc/stable/reference/generated/numpy.split.html

**Comment by mruberry** [comment]
> AFAIK, it does not have a direct name equivalent in NumPy (semantically I think it is similar to np.split with list or tuple for `indices_or_sections`)
> 
> https://numpy.org/doc/stable/reference/generated/numpy.split.html

You're totally right, sorry what I said was confusing. I mean we should check its consistency as part of resolving our split UX. See this issue for an example of how our split behavior is confusing: https://github.com/pytorch/pytorch/issues/50012

**Comment by albanD** [comment]
The nice thing is that it is listed in the view ops in the doc haha.
I think that historically, this has been a helper function to implement many of the tensor_split/chunk functions.
But we should document it indeed.

**Comment by nivesh48** [comment]
@albanD, could be assigned for this task if no one is working on this?

**Comment by albanD** [comment]
Sure.
You can add me as a reviewer for the PR when it's ready!

**Comment by nivesh48** [comment]
Sure, thank you @albanD

**Comment by nivesh48** [comment]
Hey, @albanD, Just to understand what's going that I updated the _tensor_view.rst and added the doc strings in torch for torch.split_with_sizes() and built the html on my branch, yet the documented docstring (.py) file info is not visible.
